### PR TITLE
Return all visual elements needed

### LINKED ISSUE #7027: [Bug] Tabbing through entries crashes when inside tableview
### Description
When cycling through entries that are embedded in a static `TableView`, the app will crash and throw an "Object reference not set to an instance of an object" error.

### Steps to Reproduce

1. Create a new xamarin forms app using 4.1.0.618606
2. Create a new `TableView` and add some `ViewCell`s containing an `Entry`
```
<!-- An example of such view -->
<TableView Intent="Form">
    <TableSection Title="Text Input">
        <ViewCell>
            <Entry Placeholder="Enter Text..."/>
        </ViewCell>
        <ViewCell>
            <Entry Placeholder="Enter Text..."/>
        </ViewCell>
        <ViewCell>
            <Entry Placeholder="Enter Text..."/>
        </ViewCell>
        <ViewCell>
            <Entry Placeholder="Enter Text..."/>
        </ViewCell>
    </TableSection>
</TableView>
```
3. Build and run the app on an iOS simulator
4. Click into any `Entry`
5. Press tab

### Expected Behavior
The next `Entry` in the view becomes focussed.

### Actual Behavior
System.NullReferenceException is thrown

### Basic Information
Here is the whole exception output:
Exception: System.NullReferenceException
Message: Object reference not set to an instance of an object
Source: Xamarin.Forms.Platform.iOS
Stack Trace:
 ```
at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].FocusSearch (System.Boolean forwardDirection) [0x00047] in <0ceb9d3ba964458e9e275352d39f1889>:0 
  at Xamarin.Forms.Platform.iOS.VisualElementRenderer`1[TElement].TabForward (UIKit.UIKeyCommand cmd) [0x00000] in <0ceb9d3ba964458e9e275352d39f1889>:0 
  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.110/src/Xamarin.iOS/UIKit/UIApplication.cs:86 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0000e] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.14.0.110/src/Xamarin.iOS/UIKit/UIApplication.cs:65 
  at tabtest.iOS.Application.Main (System.String[] args) [0x00001] in /Users/alex.duffell/Projects/tabtest/tabtest.iOS/Main.cs:17 
```

- Version with issue: 4.1.0.618606
- Last known good version:4.1.0.555618 (doesnt crash but doesn't select next entry)
I'm having a tough time isolating where the tab through entries stopped working. It works in my app thats running 3.1.0.697729 but won't reproduce in a sample project...
- IDE: Visual Studio For Mac 2019 v8.2.1 b19
- Platform Target Frameworks:
  - iOS:  12.4 (latest)
- Nuget Packages: Xamarin Forms Only
- Affected Devices: iOS

### Screenshots

Here is a gif of me tabbing through a `StackLayout` (this is in my provided sample code, commented out).

working gif

Here is a gif of the issue when wrapped in a `TableView`

crashing gif

### Reproduction Link

[Tab Bug Reproduction.zip](https://github.com/xamarin/Xamarin.Forms/files/3450935/Tab.Bug.Reproduction.zip)

### PR DESCRIPTION
**Pull request by Depechie**
**CORE change, guess this needs a very detailed review to see if I did not miss anything**

### Description of Change ###

Currently when you query a TableView for it's visual descendants you will get a NULL collection back.
Resolved this to return all internal elements.

### Issues Resolved ### 

- fixes #7027

### API Changes ###

Changed:
 - object TableView => Added LogicalChildrenInternal override
 - object TabIndexExtensions => Extra check for TableView

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

App will not crash and go to next element in tab queue.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

I added some entries and a stacklayout with entries to a TableView section in the Controls app. Instead of crashing it will go to the next item while tabbing.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/TableView.cs [hunk 1]
```diff
@@ -151,6 +159,17 @@ void OnTableModelRootPropertyChanged(object sender, PropertyChangedEventArgs e)
 				OnModelChanged();
 		}
 
+		private IList<Element> GetInternalChildren()
```
Comment (rmarinho) [inline Xamarin.Forms.Core/TableView.cs]: Drop private

**Comment by Depechie** [comment]
@rmarinho will for sure look at the unit test, when I have some more time later ( think next week )

**Comment by Depechie** [comment]
@rmarinho added test page, also wanted todo an UI unittest, but came to realize we can't UI test a TAB key ( or am I missing something? ).

**Comment by rmarinho** [comment]
@Depechie thanks you can add the test and mark it with category ManualReview

**Comment by rmarinho** [comment]
@Depechie I also think you can UNIT test this since all the code is on core.

**Pull request merged**